server/controllers/authenticate.js

```diff
@@ -238,25 +238,23 @@ const confirmEmailToken = async (req, res) => {
     }
 }
 
-/*  This function gets the list of available majors from a university. It requires an access token and can be used by any account type,
-    student, faculty, or industry, it takes no fields in the request body.
+/*  This function gets the list of available majors from the university given as a 
+    query parameter. It has no other requirements and can be used by any account type, 
+    including student, faculty, and industry. It takes no fields in the request body.
 */
 const getAvailableMajors = async (req, res) => {
     try {
-        //Decode Access Token
-        const accessToken = req.header('Authorization').split(' ')[1];
-        const decodeAccessToken = JWT.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+        //Get the location query parameter
+        const location = req.query.university;
+        if (!location) {
+            return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { location: location }));
```

Comment:
If the condition is !location, then I believe it will be null or undefined. In which case it won't return the location. Instead you should replace the location in the response with details and state that the location is invalid or not existing

Reply:
You are absolutely right, brain fart on my part. I will fix it.